DAGs/export_newdea_db.py

```
@@ -59,120 +60,133 @@ def __str__(self):
-                export_status = driver.find_element(
-                    By.XPATH, '//*[@id="ctl00_M_C_Content_G_ctl00__0"]/td[7]/span')
-                print('Status: ' + export_status.text)
-                time.sleep(60)
-                print('TOTAL CURRENT EXPORT WAIT TIME: {}min'.format(wait_count))
-
-                if export_status.text == 'Complete (Error)':
-                    driver.quit()
+    if login():
+        active_export = check_for_active_export()
+        no_of_checks = 1
+        while active_export:
+            if no_of_checks == 3:
+                print('Done 3 checks already. Terminating..')
+                return
+
+            print('Export ongoing, sleeping for 3 minutes...')
+            time.sleep(180)
+            active_export = check_for_active_export()
+            no_of_checks += 1
+
+        print('No ongoing export found. Starting new export process...')
+        active_export = do_export()
+        no_of_checks = 1
+        while active_export:
+            print(f'Total wait time: {no_of_checks - 1} min(s)')
+            if no_of_checks == 60:
+                print('Done 60 checks already. Terminating..')
+                raise NewdeaError('Export taking too long to complete!')
+
+            active_export, status = get_export_status()
+
+            if active_export:
+                if status == 'Complete (Error)':
                     raise NewdeaError('Export Failed!')
 
-                if wait_count == 60:
-                    driver.quit()
-                    raise NewdeaError('Export takes too long to complete!')
+                no_of_checks += 1
 
-                if export_status.text != 'Complete (Success)':
-                    wait_count += 1
-                    driver.refresh()
-                else:
-                    active_export = False
-                    driver.quit()
+            time.sleep(60)
+
+        print('Export done!')
+    else:
+        print('Unable to login')
```

@michaelbukachi if login is unsuccessful we should raise an exception indicating failure of the task; with the print statement this task will give a false positive when login fails.

cc @TAnas0